### DISCUSSION TITLE
[功能]: 话题自动命名时机应该改在话题结束时

### DISCUSSION THREAD

**Discussion by ousugo** [CherryHQ/cherry-studio#1850] [Ideas]
### 您的功能建议是否与某个问题相关?

现在的话题自动命名在第一轮对话结束时就进行了，后续就算更换了讨论的话题，话题的名称也不会发生变化了，导致话题名称可能和话题实际内容出现较大偏差。

### 请描述您希望实现的解决方案

将话题自动命名改为在当前话题结束，新开一个话题的时候再进行。这样能够保证话题名称和实际话题内容相符。

**Comment by kangfenmao** [comment]
这个默认行为可能会消耗太多 token，从而影响很多用户，让所有人的使用成本上升。

**Comment by ousugo** [comment]
> 这个默认行为可能会消耗太多 token，从而影响很多用户，让所有人的使用成本上升。

这个也的确，或者考虑给个选项？毕竟可以把命名交给一些免费模型去做

**Comment by miaotouy** [comment]
> 这个默认行为可能会消耗太多 token，从而影响很多用户，让所有人的使用成本上升。

可以参考lobe的设置话题命名的触发位置，在几轮对话后才触发命名
